src/tools/rust-demangler/src/lib.rs

```diff
@@ -0,0 +1,22 @@
+use regex::Regex;
+use rustc_demangle::demangle;
+
+const REPLACE_COLONS: &str = "::";
+
+pub fn create_disambiguator_re() -> Regex {
+    Regex::new(r"\[[a-f0-9]{5,16}\]::").unwrap()
+}
+
+pub fn demangle_lines(buffer: &str, strip_crate_disambiguators: Option<Regex>) -> Vec<String> {
+    let lines = buffer.lines();
+    let mut demangled_lines = Vec::new();
+    for mangled in lines {
+        let mut demangled = demangle(mangled).to_string();
+        if let Some(re) = &strip_crate_disambiguators {
+            demangled = re.replace_all(&demangled, REPLACE_COLONS).to_string();
+        }
+        demangled_lines.push(demangled);
+    }
+    demangled_lines.push("".to_string());
```

Comment:
It seems kind of weird to me for the library to do this. I assume this was done originally just to have `rust-demangler` emit an extra newline at the end of output? I think it would be be better to do this in the `main` function than in the library function.

Reply:
The library doesn't do this. The input string has newlines at the beginning and end, so all mangled names can be by themselves on a line. 

So I expected it to pass the blank lines through, an it worked as expected.

Reply:
Cancel that comment. I replied before reading in context.

Reply:
I guess I'm OK with that. I agree it seems weird in retrospect. It was partially done because the effect of the `lines()` method is weird; it can return the same result with different inputs (depending on trailing newline, as I recall).

I think I should just change this function to take a lines iterator as input, so the `lines()` call and its workaround (adding the newline) are in the same place in main.

Reply:
done